### PR TITLE
Created JavaScript file

### PR DESCRIPTION
**Pull request by SomyaGandhi**
Fixes #2249 

#### Checklist:
- [-] 4 space indentation.
- [-] Coding conventions are followed.
- [ ] Input is taken dynamically.
- [-] Sample Input / Output is added at the end of file.
- [-] Logic Documentation (Comments).
- [-] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- 
- 

#### Languages Used:
- JavaScript
- 

#### Files Added:
- introsort.js
- 

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

File: Introsort/introsort.js [hunk 1]
```diff
@@ -0,0 +1,121 @@
+function swap(input, left, right){
+    var temp = input[left];
+    input[left] = input[right];
+    input[right] = temp;
+}
+
+
+const insertion_Sort = (input) => {
+    for (let i = 1; i < input.length; i++) {
+        let j = i - 1
```
Comment (SakshiShreya) [inline Introsort/introsort.js:10]: Please use semi-colons consistently.

File: Introsort/introsort.js [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+            j--
+        }
+        input[j+1] = temp
+    }
+    return input;
+}
+
+
+function partition(input, left, right,pivot) {
+        i       = left, //left pointer
```
Comment (SakshiShreya) [inline Introsort/introsort.js:23]: Remove the extra spaces before `=`

File: Introsort/introsort.js [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+    if (max != i) {
+        swap(input, i, max);
+        heap_root(input, max);
+    }
+}
+
+
+function heapSort(input) {
+
```
Comment (SakshiShreya) [inline Introsort/introsort.js:64]: Remove this new line at the beginning of the function. Be consistent.

File: Introsort/introsort.js [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+        heap_root(input, max);
+    }
+}
+
+
+function heapSort(input) {
+    
+    array_length = input.length;
+
+    for (var i = Math.floor(array_length / 2); i >= 0; i -= 1)      {
```
Comment (SakshiShreya) [inline Introsort/introsort.js:67]: Remove the extra spaces between `)` and `{`.

File: Introsort/introsort.js [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+        }
+    } else {
+        if (input[hi] < input[mid]) {
+            return (input[hi] < input[lo]) ? input[lo] : input[hi];
+        } else {
+            return input[mid];
+        }
+    }
+}
+var size_threshold = 16;
```
Comment (SakshiShreya) [inline Introsort/introsort.js:95]: Leave a new-line after each function.

File: Introsort/introsort.js [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+        }
+        depth_limit=depth_limit-1;
+        var p = partition(input, lo, hi, medianof3(input, lo, lo+((hi-lo)/2)+1, hi-1));
+        introsort_loop(input, p, hi, depth_limit);
+        hi = p;
+    }
+    insertion_Sort(input, lo, hi);
+}
+
+var arr = [-1, 3, 6, 6, 5, 4, -9 ,9, 0, 2, 4, 1, 6, 7 ];
```
Comment (SakshiShreya) [inline Introsort/introsort.js:111]: Fix the space after `-9`

File: Introsort/introsort.js [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+introsort_loop(arr,0,arr.length, (Math.log(arr.length))*2);
```
Comment (SakshiShreya) [inline Introsort/introsort.js:112]: Add space after each comma.

File: Introsort/introsort.js [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+console.log(arr);
+
+/*Output
+[
+  -9, -1, 0, 1, 2, 3,
+   4,  4, 5, 6, 6, 6,
+   7,  9
+]
+ */
```
Comment (SakshiShreya) [inline Introsort/introsort.js:121]: Add a new-line at the EOF.

**Comment by SakshiShreya** [comment]
Also fix the PR template and filename.

**Comment by somya-kapoor** [comment]
Are you working on it?

**Comment by somya-kapoor** [comment]
Due to no activity for more than 7 days, I am closing this issue.

**Pull request closed without merging**